### PR TITLE
format simple constructor applies on the same line

### PR DESCRIPTION
**Pull request by V-FEXrt**
An apply that "looks" like a constructor, something like 

```
Identifier (
  some,
  expression
)
```

is now formatted without a newline between the lhs and rhs

Bonus: the assign operator (`:->` )  is always emitted "flat" as adding newlines makes the code confusing.

Bonus #2: if-else-match has else and match on the same line

### CODE REVIEW

**Comment by V-FEXrt** [comment]
This has been cluttering the repo for a bit. going to close for now and just reference the closed PR

File: tests/wake-format/basic/stdout [hunk 1]
```diff
@@ -982,3 +981,61 @@ export def !(x: Boolean): Boolean =
...
+        def imp m p c = prim "write"
+        def pre input = Pair input Unit
+        def post = match _
+            Pair (Fail f) _ = Fail f
+            Pair (Pass output) Unit =
+                if mode < 0 || mode > 0x1ff then
+                    Fail (makeError "write {path}: Invalid mode ({strOctal mode})")
+                else match (imp mode path content)
+                    Fail f = Fail (makeError f)
+                    Pass path = Pass (editRunnerOutputOutputs (path, _) output)
```
Comment (ag-eitilt) [inline tests/wake-format/basic/stdout:1007]: Are these the constructors the main PR was targeting?  I don't see much else other than the initial `emccFn (` which seems to be exercised, though I've also not looked at the C++ to know *what* to look for here.  Nevertheless, looks good!

**Review by ag-eitilt — APPROVED** [review]

**Review by JakeSiFive — APPROVED** [review]
C++ code looks good to me.

File: tests/wake-format/basic/stdout (hunk 1 above)
Comment (V-FEXrt) [reply tests/wake-format/basic/stdout:1007]: No, I think the test here was the "else match" part. Though these constructors not getting exploded does show a bit of that functionality

**Pull request merged**